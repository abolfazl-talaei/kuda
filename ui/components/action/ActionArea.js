import React from 'react';
import { Button, Card, Jumbotron } from 'react-bootstrap';
import styles from './ActionArea.module.css';

const ActionArea = (props) => {
  const { title, description, actionText, action, url, direction } = props;

  return (
    <Card style={{ textAlign: direction }} className={styles.container}>
      <Card.Body className={styles.cardBody}>
        <Jumbotron style={{ textAlign: direction }}>
          <h1>{title}</h1>
          <p>{description}</p>
          <p>
            {action && (
              <Button variant="primary" onClick={() => action()}>
                {actionText}
              </Button>
            )}
            <Button variant="primary" href={url} className={styles.button}>
              {actionText}
            </Button>
          </p>
        </Jumbotron>
      </Card.Body>
    </Card>
  );
};

export default ActionArea;
