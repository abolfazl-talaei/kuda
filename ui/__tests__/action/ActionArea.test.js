import { configure, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import ActionArea from '../../components/action/ActionArea';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Button } from 'react-bootstrap';

test('<ActionArea />', () => {
  configure({ adapter: new Adapter() });

  const mockCallBack = jest.fn();

  const component = (
    <ActionArea
      className="m-3"
      title="test"
      description="test"
      actionText="test"
      url="newkuda"
      direction="right"
      action={mockCallBack}
    ></ActionArea>
  );
  const componentRender = renderer.create(component);
  let tree = componentRender.toJSON();
  expect(tree).toMatchSnapshot();

  const componentShallow = shallow(component);
  componentShallow.children().at(0).simulate('click');
});
