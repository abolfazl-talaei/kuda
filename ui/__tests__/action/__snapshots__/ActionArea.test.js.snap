// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionArea /> 1`] = `
<div
  className="card"
  style={
    Object {
      "textAlign": "right",
    }
  }
>
  <div
    className="card-body"
  >
    <div
      className="jumbotron"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <h1>
        test
      </h1>
      <p>
        test
      </p>
      <p>
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          test
        </button>
        <a
          className="btn btn-primary"
          href="newkuda"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          test
        </a>
      </p>
    </div>
  </div>
</div>
`;
